src/test/java/org/springframework/integration/aws/inbound/S3InboundChannelAdapterTests.java

```diff
@@ -191,7 +191,7 @@ public S3InboundFileSynchronizer s3InboundFileSynchronizer() {
 			synchronizer.setPreserveTimestamp(true);
 			synchronizer.setRemoteDirectory(S3_BUCKET);
 			synchronizer.setFilter(new S3RegexPatternFileListFilter(".*\\.test$"));
-			Expression expression = PARSER.parseExpression("#this.toUpperCase() + '.a'");
+			Expression expression = PARSER.parseExpression("(#this.contains('/') ? #this.substring(#this.lastIndexOf('/') + 1) : #this).toUpperCase() + '.a'");
```

Comment:
Thank you, @riversidetraveler , for the contribution!

Looks like we have here one more issue.
I formed it into the https://jira.spring.io/browse/INT-4086 because there is nothing to do on the SI-AWS side.